bloom-housing/bloom
# fix: add home type feature to backend and partners

This PR addresses #4432, #4431 


## Description

Adds the "Home Type" select option to the listing create/edit page as well as connecting HomeType to the listing create endpoint, edit endpoint, and the CSV export.

This is also the first feature using the new feature flag table so some extra logic to utilize it has been added

NOTES:
- I did not get approval from product/design on the location of this field in Partners. The Detroit location is slightly different because it is within the "Unit Group" section which doesn't exist in core. I will get confirmation on the location before merging of this PR. 

## How Can This Be Tested/Reviewed?

- Reseed the database with staging data. This will add the new feature flag to the "Bloomington" jurisdiction
- Go to partner site and create a new listing
- Select Bloomington Jurisdiction and add a home type to the listing
- Save listing and the home type should appear on the listing
- Edit listing and change the home type. After save it should appear
- Export the Listing CSV and the "Home Type" column should be there with the new listing's home type in the appropriate location

Other things you can test:
- Verify the Home Type field doesn't appear on the UI if you have selected a different jurisdiction
- If no jurisdiction has the home type feature flag turned on the CSV export should not have that column

## Author Checklist:


## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates
